sparkler-api/src/main/java/edu/usc/irds/sparkler/Fetcher.java

```diff
@@ -17,10 +17,13 @@
 
 package edu.usc.irds.sparkler;
 
+import edu.usc.irds.sparkler.model.FetchedData;
+
 /**
  * Created by thammegr on 6/21/16.
  */
 public interface Fetcher extends ExtensionPoint {
 
-    //TODO: this is incomplete
+	FetchedData fetch(String url);
```

Comment:
How about:

```
Iterator<FetchedData> fetch(Iterator<String> urls);
```
The idea is that, fetcher may accept stream of URLS, return stream of contents one after the other. It insert fetch delay in between based on the configuration.

I see this is efficient and convenient.

Efficient:  create a single browser/driver/httpclient object and reuse for all fetches one after the other in the stream.
Convenient: When authentication is required, the fetcher can login once, and fetch all the pages in the same host one after the other and logout once it is done(assumption:the urls are grouped by host) . That way we dont get blocked due to too many authentication requests.